tensorflow_quantum/core/qsim/unitary_space_slow.cc

```diff
@@ -0,0 +1,356 @@
+              data[2 * p * sizei + 2 * pp] =
+                  dot_two_r(m01_r, m01_i, m11_r, m11_i, m21_r, m21_i, m31_r,
+                            m31_i, 2, matrix);
+              data[2 * p * sizei + 2 * pp + 1] =
+                  dot_two_i(m01_r, m01_i, m11_r, m11_i, m21_r, m21_i, m31_r,
+                            m31_i, 2, matrix);
+              pp = si2 | sizej;
+              data[2 * p * sizei + 2 * pp] =
+                  dot_two_r(m02_r, m02_i, m12_r, m12_i, m22_r, m22_i, m32_r,
+                            m32_i, 2, matrix);
+              data[2 * p * sizei + 2 * pp + 1] =
+                  dot_two_i(m02_r, m02_i, m12_r, m12_i, m22_r, m22_i, m32_r,
+                            m32_i, 2, matrix);
+              pp |= sizek;
+              data[2 * p * sizei + 2 * pp] =
+                  dot_two_r(m03_r, m03_i, m13_r, m13_i, m23_r, m23_i, m33_r,
+                            m33_i, 2, matrix);
+              data[2 * p * sizei + 2 * pp + 1] =
+                  dot_two_i(m03_r, m03_i, m13_r, m13_i, m23_r, m23_i, m33_r,
+                            m33_i, 2, matrix);
+
+              p |= sizek;
+              pp = si2;
+              data[2 * p * sizei + 2 * pp] =
+                  dot_two_r(m00_r, m00_i, m10_r, m10_i, m20_r, m20_i, m30_r,
+                            m30_i, 3, matrix);
+              data[2 * p * sizei + 2 * pp + 1] =
+                  dot_two_i(m00_r, m00_i, m10_r, m10_i, m20_r, m20_i, m30_r,
+                            m30_i, 3, matrix);
+              pp = si2 | sizek;
+              data[2 * p * sizei + 2 * pp] =
+                  dot_two_r(m01_r, m01_i, m11_r, m11_i, m21_r, m21_i, m31_r,
+                            m31_i, 3, matrix);
+              data[2 * p * sizei + 2 * pp + 1] =
+                  dot_two_i(m01_r, m01_i, m11_r, m11_i, m21_r, m21_i, m31_r,
+                            m31_i, 3, matrix);
+              pp = si2 | sizej;
+              data[2 * p * sizei + 2 * pp] =
+                  dot_two_r(m02_r, m02_i, m12_r, m12_i, m22_r, m22_i, m32_r,
+                            m32_i, 3, matrix);
+              data[2 * p * sizei + 2 * pp + 1] =
+                  dot_two_i(m02_r, m02_i, m12_r, m12_i, m22_r, m22_i, m32_r,
+                            m32_i, 3, matrix);
+              pp |= sizek;
+              data[2 * p * sizei + 2 * pp] =
+                  dot_two_r(m03_r, m03_i, m13_r, m13_i, m23_r, m23_i, m33_r,
+                            m33_i, 3, matrix);
+              data[2 * p * sizei + 2 * pp + 1] =
+                  dot_two_i(m03_r, m03_i, m13_r, m13_i, m23_r, m23_i, m33_r,
+                            m33_i, 3, matrix);
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+tensorflow::Status UnitarySpaceSlow::ApplyGate1(const float* matrix) {
+  float mat_00_r, mat_01_r, mat_10_r, mat_11_r;
+  float mat_00_i, mat_01_i, mat_10_i, mat_11_i;
+
+  auto state = GetRawUnitary();
+  mat_00_r = dot_one_r(state[0], state[1], state[4], state[5], 0, matrix);
```

Comment:
Would be clearer to make the unitary be called `data` as above

Reply:
Done.